/*
 * This file is part of the NUbots Codebase.
 *
 * The NUbots Codebase is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * The NUbots Codebase is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with the NUbots Codebase.  If not, see <http://www.gnu.org/licenses/>.
 *
 * Copyright 2013 NUBots <nubots@nubots.net>
 */

syntax = "proto3";

package message.behaviour.proto;

message Subsumption {

    enum Limb {
        UNKNOWN = 0;
        LEFT_LEG = 1;
        RIGHT_LEG = 2;
        LEFT_ARM = 3;
        RIGHT_ARM = 4;
        HEAD = 5;
    }

    message LimbSet {
        float priority = 1;
        repeated Limb limbs = 2;
    }

    message ActionRegister {
        uint32 id = 1;
        string name = 2;
        repeated LimbSet limb_set = 3;
    }

    message ActionStateChange {
        enum State {
            UNKNOWN = 0;
            START = 1;
            KILL = 2;
        }

        State state = 1;
        string name = 2;
        repeated Limb limbs = 3;
    }

    message ActionPriorites {
        uint32 id = 1;
        repeated float priorities = 2;
    }

    repeated ActionRegister action_register = 1;
    repeated ActionStateChange action_state_change = 2;
    repeated ActionPriorites action_priority_change = 3;
}
